tests/integration/data_sources_and_expectations/test_cloud_data_source_types.py

```diff
@@ -0,0 +1,87 @@
+from typing import Generator
+
+import pandas as pd
+import pytest
+from sqlalchemy.orm import MappedColumn
+
+from great_expectations.compatibility import snowflake
+from great_expectations.compatibility.sqlalchemy import sqltypes
+from great_expectations.datasource.fluent.interfaces import Batch
+from great_expectations.expectations import ExpectColumnSumToBeBetween
+from tests.integration.test_utils.data_source_config import SnowflakeDatasourceTestConfig
+from tests.integration.test_utils.data_source_config.snowflake import SnowflakeBatchTestSetup
+
+
+class TestSnowflakeTypes:
+    """This set of tests ensures that we can run expectations against every data
+    type supported by Snowflake.
+
+    https://docs.snowflake.com/en/sql-reference/intro-summary-data-types
+    """
+
+    COLUMN = "col_a"
+
+    @pytest.mark.snowflake
+    @pytest.mark.parametrize(
+        "data_type",
+        [
+            snowflake.NUMBER,
+            sqltypes.NUMERIC,
+            sqltypes.INT,
+            sqltypes.BIGINT,
+            sqltypes.SMALLINT,
+            sqltypes.FLOAT,
+            sqltypes.DOUBLE,
+            sqltypes.DOUBLE_PRECISION,
+            sqltypes.REAL,
+        ],
+    )
+    def test_numeric(self, data_type):
+        batch = self.setup_batch(
+            data=pd.DataFrame({self.COLUMN: [1, 2, 3, 4]}),
+            column_type=data_type,
+        )
+        result = batch.validate(
+            expect=ExpectColumnSumToBeBetween(
+                column=self.COLUMN,
+                min_value=9,
+                max_value=11,
+            )
+        )
+        assert result.success
+
+    def test_string(self): ...
+
+    def test_boolean(self): ...
+
+    def test_date(self): ...
+
+    def test_semi_structured(self): ...
+
+    def test_geospatial(self): ...
+
+    def test_vector(self): ...
+
+    def setup_batch(self, data: pd.DataFrame, column_type: MappedColumn) -> Batch:
```

Comment:
How does this not being a fixture affect setup/teardown? pytest does some magic that we take advantage of, e.g. how we use that `_cleanup` fixture. Would moving this out to a fixture solve this? Or is this not even a problem?

I like the high level. This would have been tedious if we only used the other decorator based on running things with many datasources.